### DISCUSSION TITLE
Are the asserts about self.buffer().is_some() really necessary in src/macros/display.rs?

### DISCUSSION THREAD

**Discussion by degaart** [fltk-rs/fltk-rs#1204] [General]
Hello,

Using fluid and fl2rust, TextDisplay widgets where text properties like font, font color, etc... hit these asserts, as the generated code from fl2rust do not set the TextDisplay's buffer before setting properties.

A closed issue (https://github.com/fltk-rs/fltk-rs/issues/752#issuecomment-870765144) states that these asserts are unnecessary and were removed, yet they are still present in master.

**Comment by MoAlyousef** [comment]
Hello

Some TextDisplay widget text-related methods check for the presence of a text buffer as defensive mechanism. Whether these methods really require a text buffer or not needs to be checked individually. I haven’t had the time to check them. If you have tried such methods without the presence of a text buffer (by removing the assert) and things worked out, you can report here, or you can also create a pull request if you prefer. Otherwise I’ll try to check these methods in the coming days.

**Reply by degaart** [reply]
Hello,

I sent a pull request, but as I had to add some stuff to the cfltk submodule, I worry you won't be able to merge cleanly.

**Reply by MoAlyousef** [reply]
Sorry for the late reply. I'll take a look now at the pull request.
